kratos/utilities/mortar_utilities.h

```diff
@@ -766,95 +755,85 @@ class MortarUtilities
 template<> 
 inline void MortarUtilities::ResetValue<Variable<double>, Historical>(
         ModelPart& rThisModelPart,
-        Variable<double>& ThisVariable, 
-        const bool InvertedPair
-        ) {
+        Variable<double>& ThisVariable
+        ) 
+{
     NodesArrayType& nodes_array = rThisModelPart.Nodes();
     
     // We set to zero
     #pragma omp parallel for
     for(int i = 0; i < static_cast<int>(nodes_array.size()); ++i) {
         auto it_node = nodes_array.begin() + i;
-        if (it_node->Is(SLAVE) == !InvertedPair) 
-            it_node->FastGetSolutionStepValue(ThisVariable) = 0.0;
+        it_node->FastGetSolutionStepValue(ThisVariable) = 0.0;
     }
 }
 
 template<> 
 inline void MortarUtilities::ResetValue<ComponentType, Historical>(
         ModelPart& rThisModelPart,
-        ComponentType& ThisVariable, 
-        const bool InvertedPair
-        ) {
+        ComponentType& ThisVariable
+        ) 
+{
     NodesArrayType& nodes_array = rThisModelPart.Nodes();
     
     // We set to zero
     #pragma omp parallel for
     for(int i = 0; i < static_cast<int>(nodes_array.size()); ++i) {
         auto it_node = nodes_array.begin() + i;
-        if (it_node->Is(SLAVE) == !InvertedPair) 
-            it_node->FastGetSolutionStepValue(ThisVariable) = 0.0;
+        it_node->FastGetSolutionStepValue(ThisVariable) = 0.0;
     }
 }
 
 template<> 
 inline void MortarUtilities::ResetValue<Variable<array_1d<double, 3>>, Historical>(
```

Comment:
Actually I think you can remove all of the `ResetValue` functions and replace them with the function from the `VariableUtils`